aviatrix/data_source_aviatrix_firewall_instance_images.go

```
@@ -73,3 +80,140 @@ func dataSourceAviatrixFirewallInstanceImagesRead(d *schema.ResourceData, meta i
+					}
+				} else {
+					//format: Rab-xxx.xxx
+					splitFlag = "-"
+					if compareVersion2(listVersion[i], listVersion[j], splitFlag) == 2 {
+						listVersion[i], listVersion[j] = listVersion[j], listVersion[i]
+					}
+				}
+			} else if checkFirstCharacter(listVersion[i]) == "P" {
+				//format: Pa-bc-xx.xx.xx
+				splitFlag = "-"
+				if compareVersion3(listVersion[i], listVersion[j], splitFlag) == 2 {
+					listVersion[i], listVersion[j] = listVersion[j], listVersion[i]
+				}
+			} else {
+				//format: xx.xxx.xxx
+				splitFlag = "."
+				if compareVersion(listVersion[i], listVersion[j], splitFlag) == 2 {
+					listVersion[i], listVersion[j] = listVersion[j], listVersion[i]
+				}
+			}
+		}
+	}
+	return listVersion
+}
+
+func checkFirstCharacter(input string) string {
+	firstCharacter := input[0:1]
+	return firstCharacter
+}
+
+//format: xx.xx.xx
+func compareVersion(version1, version2, splitFlag string) (res int) {
+	imageVersionArray1 := strings.Split(version1, splitFlag)
+	imageVersionArray2 := strings.Split(version2, splitFlag)
+	for index := range imageVersionArray1 {
+		reg, _ := regexp.Compile("[^0-9]+")
+		v1SliceString := reg.ReplaceAllString(imageVersionArray1[index], ".")
+		v2SliceString := reg.ReplaceAllString(imageVersionArray2[index], ".")
+		int1, _ := strconv.ParseFloat(v1SliceString, 32)
+		int2, _ := strconv.ParseFloat(v2SliceString, 32)
+		if int1 > int2 {
+			return 1
+		}
+		if int1 < int2 {
+			return 2
+		}
+	}
+	return 1
+}
+
+//format: Rab-xxx.xxx and Rab_abcxx.x
+func compareVersion2(version1, version2, flag string) (res int) {
+	imageVersionArray1 := strings.Split(version1, flag)
+	imageVersionArray2 := strings.Split(version2, flag)
+	for index := range imageVersionArray1 {
+		reg, _ := regexp.Compile("[^0-9.]+")
+		v1SliceString := reg.ReplaceAllString(imageVersionArray1[index], "")
+		v2SliceString := reg.ReplaceAllString(imageVersionArray2[index], "")
+		int1, _ := strconv.ParseFloat(v1SliceString, 32)
```

Is it a good idea to ignore the error here? Can we assume that this will always succeed?

---

i will add the error log here, thank you